src/main/java/org/prebid/server/bidder/compass/CompassBidder.java

```diff
@@ -0,0 +1,168 @@
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpCompass extImpCompass) {
+        final CompassImpExtBidder compassImpExtBidderWithType = getImpExtCompassWithType(extImpCompass);
+        final ObjectNode modifiedImpExtBidder = mapper.mapper().createObjectNode();
+
+        modifiedImpExtBidder.set("bidder", mapper.mapper().convertValue(compassImpExtBidderWithType, JsonNode.class));
+
+        return imp.toBuilder()
+                .ext(mapper.mapper().valueToTree(modifiedImpExtBidder))
+                .build();
+    }
+
+    private CompassImpExtBidder getImpExtCompassWithType(ExtImpCompass extImpCompass) {
+        final CompassImpExtBidder.CompassImpExtBidderBuilder impExtCompass = CompassImpExtBidder.builder();
+
+        if (StringUtils.isNotEmpty(extImpCompass.getEndpointId())) {
+            impExtCompass
+                    .type("network")
+                    .endpointId(extImpCompass.getEndpointId());
+        } else if (StringUtils.isNotEmpty(extImpCompass.getPlacementId())) {
+            impExtCompass
+                    .type("publisher")
+                    .placementId(extImpCompass.getPlacementId());
+        }
+
+        return impExtCompass.build();
+    }
+
+    private HttpRequest<BidRequest> createSingleRequest(Imp imp, BidRequest request) {
+        final BidRequest outgoingRequest = request.toBuilder().imp(Collections.singletonList(imp)).build();
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
+                .build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse,
+                                               List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse, errors);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse,
```

Comment:
Nitpick: merge `extractBids` and `bidsFromResponse` in one method (`extractBids`).